src/test/unit/add.test.ts

```diff
@@ -194,6 +194,44 @@ describe('Add Command', () => {
       )
     })
 
+    it('passes metadata options through to upload and storage context', async () => {
+      await runAdd({
+        filePath: testFile,
+        privateKey: 'test-private-key',
+        rpcUrl: 'wss://test.rpc.url',
+        metadata: { region: 'us-west', note: '' },
+        dataSetMetadata: { purpose: 'erc8004' },
+      })
+
+      const { performUpload } = await import('../../common/upload-flow.js')
+      expect(vi.mocked(performUpload)).toHaveBeenCalledWith(
+        expect.anything(),
+        expect.anything(),
+        expect.anything(),
+        expect.objectContaining({
+          metadata: { region: 'us-west', note: '' },
+        })
+      )
+
+      const { createStorageContext, initializeSynapse } = await import('../../core/synapse/index.js')
+      expect(vi.mocked(createStorageContext)).toHaveBeenCalledWith(
+        expect.anything(),
+        expect.anything(),
+        expect.objectContaining({
+          dataset: {
+            metadata: { purpose: 'erc8004' },
+          },
+        })
+      )
+
+      expect(vi.mocked(initializeSynapse)).toHaveBeenCalledWith(
+        expect.objectContaining({
```

Comment:
JS newbie: is there a reason we check the synapse initialization at the end?  I would have thought to check it first since it happens first.

Reply:
fixed and put them in order